test/test_user_config.js

```diff
@@ -0,0 +1,299 @@
+/**
+ * Unit tests for user configuration structures.
+ *
+ * @license BSD-3-Clause
+ */
+
+import {FilterSet} from "user_config";
+
+function buildUserConfigTests() {
+  QUnit.module("FilterSet", function () {
+    const makeBasicExample = () => {
+      return new FilterSet(
+        2020,
+        "bau",
+        "commercial",
+        "HFC-134a",
+        "sales:import:mt / yr",
+        "simulations",
+        "baseline",
+        true,
+      );
+    };
+
+    const makeNullExample = () => {
+      return new FilterSet(null, null, null, null, null, null, null, null);
```

Comment:
Apologies… null should not be a valid value for attributeImporter. Please have this use false for the final parameter in the null example. Please update tests and the JSDoc in user_config accordingly.